### PR TITLE
sub ReadonlyReplica metric when detach readonly tables

### LINKED ISSUE #15598: A table keep turning into read-only mode
I am facing a strange behavior with ClickHouse.

One of the replicas keep turning into readonly mode. I am detecting this by pinging the server as follows:
```
$ curl localhost:8123/replicas_status

Code: 242, e.displayText() = DB::Exception: Table is in readonly mode (version 20.5.2.7 (official build))
```

The strange behavior is that when I query `system.metrics` table I found that `ReadonlyReplica` metric have value of 0.

```
SELECT
    metric,
    value
FROM system.metrics
WHERE metric = 'ReadonlyReplica'

┌─metric──────────┬─value─┐
│ ReadonlyReplica │     0 │
└─────────────────┴───────┘

1 rows in set. Elapsed: 0.003 sec.
```

And even more, I can do `INSERT` queries to this replica without problems.

When I restart the server and ping it, it returns `Ok`. But after couple of hours, it fall back to the same behavior.

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Bug Fix

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Decrement the `ReadonlyReplica` metric when detaching read-only tables. This fixes https://github.com/ClickHouse/ClickHouse/issues/15598

Detailed description / Documentation draft:

In our production env, `clickhouse` 20.3.8.53,  `Readonly` will not be decreased when we detach the readonly tables,  since we can't mutate `system.metrics`, the only way to handle it is to restart the whole  CH server or stop the metric alert rules.
```
ubuntu :) select * from system.replicas where is_readonly = 1\G

SELECT *
FROM system.replicas
WHERE is_readonly = 1

Ok.

0 rows in set. Elapsed: 5.119 sec. Processed 2.14 thousand rows, 938.52 KB (417.87 rows/s., 183.35 KB/s.)

ubuntu :) select * from system.metrics where metric like '%Readonly%';

SELECT *
FROM system.metrics
WHERE metric LIKE '%Readonly%'

┌─metric──────────┬─value─┬─description────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│ ReadonlyReplica │     6 │ Number of Replicated tables that are currently in readonly state due to re-initialization after ZooKeeper session loss or due to startup without ZooKeeper configured. │
└─────────────────┴───────┴────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘

```

Information about CI checks: https://clickhouse.tech/docs/en/development/continuous-integration/

#15598

### CODE REVIEW

**Comment by sundy-li** [comment]
fixes #15598

**Review by Avogar — DISMISSED** [review]

File: tests/integration/test_system_metrics/test.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        cluster.shutdown()
+
+def test_readonly_metrics(start_cluster):
+    assert node1.query("SELECT value FROM system.metrics WHERE metric = 'ReadonlyReplica'") == "0\n"
+
+    with PartitionManager() as pm:
+        ## make node1 readonly -> heal -> readonly -> heal -> detach table -> heal -> attach table
+        pm.drop_instance_zk_connections(node1)
+        zk_timeout = 300
+        for i in range(zk_timeout):
```
Comment (tavplubix) [inline tests/integration/test_system_metrics/test.py:47]: FYI https://github.com/ClickHouse/ClickHouse/blob/9cb3c743bd0ec28942b018c5853accfc1b1259b0/tests/integration/helpers/test_tools.py#L46

Comment (Avogar) [reply tests/integration/test_system_metrics/test.py:47]: Thanks!

**Review by Avogar — APPROVED** [review]

**Comment by filimonov** [comment]
Does it also solve https://github.com/ClickHouse/ClickHouse/issues/16045 ?

**Comment by alexey-milovidov** [comment]
@filimonov Does not look like.

**Comment by sundy-li** [comment]
> Does it also solve #16045 ?

No, `restarting_thread` is not started when the table is read-only during startup.

To solve that we should also have the metric code to `StorageReplicatedMergeTree.cpp`.

**Pull request merged**